packages/state-manager/src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -226,17 +243,29 @@ export const currentChannelMessagesMergedBySender = createSelector(
         const index = merged.length && merged.length - 1
         const last = merged[index][0]
 
+        // Determine if a message is "unsent"
+        const isRecent = lastConnectedTime < message.createdAt
+        const communityHasPeers = communityPeerList != null && communityPeerList.length > 1
+        const hasConnectedPeers = connectedPeers.length > 0
+        const peersDisconnectedRecently = allPeersDisconnectedAt != null && allPeersDisconnectedAt < message.createdAt
+        const noPeersThisSession = allPeersDisconnectedAt == null && connectedPeers.length > 0
+        const isUnsent =
```

Comment:
I can see a very similar code in BasicMessage. One could easily miss one of those places when updating the 'unsent' logic. Maybe we could refactor it to use the same function? It can also be easier testable this way.